### PR TITLE
Add handling for Enum(Case)s in NameResolver

### PR DESCRIPTION
**Pull request by bwoebi**

### CODE REVIEW

**Comment by nikic** [comment]
Can you please also add them to the test at https://github.com/nikic/PHP-Parser/blob/c35cc4b2cb82fd543444d6afeb42f858b5e375e3/test/PhpParser/NodeVisitor/NameResolverTest.php#L180?

**Comment by bwoebi** [comment]
Sure, done - and added the missing type printing to the pretty printer at the same time

File: lib/PhpParser/PrettyPrinter/Standard.php [hunk 1]
```diff
@@ -731,6 +731,7 @@ protected function pStmt_Enum(Stmt\Enum_ $node) {
         return $this->pAttrGroups($node->attrGroups)
              . 'enum ' . $node->name
              . (!empty($node->implements) ? ' implements ' . $this->pCommaSeparated($node->implements) : '')
+			 . ($node->scalarType ? " : $node->scalarType" : '')
```
Comment (nikic) [inline lib/PhpParser/PrettyPrinter/Standard.php:734]: Tabs...

File: test/PhpParser/NodeVisitor/NameResolverTest.php [hunk 2]
```diff
@@ -327,6 +339,7 @@ public function testAddDeclarationNamespacedName() {
             ]),
             new Stmt\Trait_('E'),
             new Expr\New_(new Stmt\Class_(null)),
+			new Stmt\Enum_('F'),
```
Comment (nikic) [inline test/PhpParser/NodeVisitor/NameResolverTest.php:342]: ...tabs

**Comment by bwoebi** [comment]
Sorry for the tabs … I wished PHPStorm would just properly recognize the indentation style used in a file and not try to put tabs inmidst of spaces :-/

**Comment by nikic** [comment]
Looks like test/code/prettyPrinter/stmt/enum.test needs an update now that the type is printed.

**Comment by bwoebi** [comment]
Oh, there are also non-PHP files, fixed.

**Pull request merged**